main.tf

```diff
@@ -707,3 +712,25 @@ resource "aws_s3_bucket_ownership_controls" "this" {
     aws_s3_bucket.this
   ]
 }
+
+resource "aws_s3_bucket_intelligent_tiering_configuration" "tiering_conf" {
+  for_each = local.intelligent_tiering_config
+  name     = each.key
+  bucket   = aws_s3_bucket.this[0].id
+  status   = each.value.status
```

Comment:
Please make `status` to accept boolean `true` / `false` also. See other resources in this module.